### ISSUE TITLE
babel-preset-expo doesn't support Babel 7 and RN 0.56

### ISSUE THREAD

**Issue by Hypnosphi** [expo/expo#1954]
In version `0.56`, React Native has migrated to Babel 7. That means that dependencies in `babel-preset-expo` need to be updated, see https://github.com/facebook/react-native/releases/tag/v0.56.0

**Comment by jesseruder** [comment]
Hi, we haven't upgraded to the new version of React Native yet. We'll take care of this when we upgrade.

**Comment by bogdan-calapod** [comment]
I think this issue is causing a problem with my project - a fresh CRNA project with Jest installed. Metro somehow depends on `@babel/core`:

```
$: npm list @babel/core
REDACTED@0.0.1 /mnt/c/Users/REDACTED
└─┬ react-native@0.55.4
  └─┬ metro@0.30.2
    └── @babel/core@7.0.0-rc.1
```

My package.json doesn't include `@babel/core` anywhere, but when running tests I get the following error in Jest:

```
  ● Test suite failed to run

    Plugin/Preset files are not allowed to export objects, only functions. In /mnt/c/REDACTED/node_modules/babel-preset-expo/index.js
```

The thing is, if I remove the `node_modules/@babel/core` folder, it automatically picks up babel v6, and it works without hiccups. A couple weeks ago it worked just fine, but this week when I ran `npm install` the problem surfaced.

I see that the `metro@0.30.2` [package.json](https://github.com/facebook/metro/blob/f2b6232c5d93cae580e88fc6741ee87d880318c8/packages/metro/package.json) indeed includes `@babel/core 7.0.0-beta` as a dependency.

Any idea how I can force Jest to ignore babelv7 installed by metro ? At the moment I've found a quirky workaround by removing the `node_modules/@babel/core` folder in CI before running tests, but I can't do this locally as the packager doesn't start without the v7 babel.

**Comment by cur33** [comment]
@bogdan-calapod Thanks for taking note of that solution; however, it still caused some dependency issues for me (with just the skeleton created by the crna tool). I fixed it by just reverting to react-native@0.55.4 for now (for anyone else who doesn't need 0.56 yet).

**Comment by MrCheater** [comment]
+1

**Comment by daviskoh** [comment]
still an issue

**Comment by matt-paul** [comment]
This issue is surfacing for me when trying to use Quokka

**Comment by nickjuntilla** [comment]
I'm still getting this error

**Comment by jkobject** [comment]
It seems I have a similar issue when trying to use [docz](www.docz.site) on my project... if someone has a solution in how to handle the dependancies. it seems that I have ```Error: Plugin/Preset files are not allowed to export objects, only functions```

**Comment by MrCheater** [comment]
@jkobject  https://blog.expo.io/expo-sdk-v31-0-0-is-now-available-cad6d0463f49
